WPFHospitalEditor/RoomScheduledExaminations.xaml.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using WPFHospitalEditor.Controller;
+using WPFHospitalEditor.Controller.Interface;
+using WPFHospitalEditor.DTOs;
+using WPFHospitalEditor.Model;
+
+namespace WPFHospitalEditor
+{
+    /// <summary>
+    /// Interaction logic for RoomScheduledAppointments.xaml
+    /// </summary>
+    public partial class RoomScheduledExaminations : Window
+    {
+        private IExaminationServerController examinationServerController = new ExaminationServerController();
+
+        private int roomId;
+        public ObservableCollection<RoomScheduledAppointmentDto> roomExaminations { get; set; }
+        public ObservableCollection<RoomScheduledAppointmentDto> roomRelocations { get; set; }
+        public ObservableCollection<RoomScheduledAppointmentDto> roomRenovations { get; set; }
+
+        public RoomScheduledExaminations(int roomId)
+        {
+            InitializeComponent();
+            this.roomId = roomId;
+            FillObservableCollection();
+            SetExaminationIdsComboBox();
+            this.DataContext = this;
+        }
+
+        private void CloseClick(object sender, RoutedEventArgs e)
+        {
+            this.Close();
+        }
+
+        private void CancelAppointment(object sender, RoutedEventArgs e)
+        {
+            if (appointmentCancelComboBox.SelectedIndex != 0)
+            {
+                string cancel = examinationServerController.Cancel(int.Parse(appointmentCancelComboBox.SelectedItem.ToString()));
+                if (!cancel.Equals("BadRequest"))
+                {
+                    MessageBox.Show("Examination is successfuly CANCELED!");
+                    ClearObservableCollectionsAndComboBox(int.Parse(appointmentCancelComboBox.SelectedItem.ToString()));
+                    SetExaminationIdsComboBox();
+                }
+                else
+                    MessageBox.Show("Examination cannot be CANCELED!");
+            } else
+                MessageBox.Show("You didn't pick any examination id!");
+        }
+
+        private void ClearObservableCollectionsAndComboBox(int examinationId)
+        {
+            RemoveDeletedExamination(examinationId);
+            appointmentCancelComboBox.Items.Clear();
+            appointmentCancelComboBox.Items.Add("None");
+            appointmentCancelComboBox.SelectedIndex = 0;
+        }
+
+        private void RemoveDeletedExamination(int examinationId)
+        {
+            if (FindAndDeleteByExaminationId(roomExaminations, examinationId)) return;
+            if (FindAndDeleteByExaminationId(roomRelocations, examinationId)) return;
+            FindAndDeleteByExaminationId(roomRenovations, examinationId);
```

Comment:
Maybe to first check renovations and relocations, just for performances. If you are canceling renovation in this case you need to pass through all normal examinations before canceling renovation.

Reply:
Fixed.